### PR TITLE
Updates to persistent storage security

### LINKED ISSUE #2264: no defined concept of related or unrelated documents
In rs:
> Reading systems that allow users to store data MUST ensure they do not make that data available to other unrelated documents

I think "unrelated" is undefined; the spec makes clear that there isn't a reliable way to determine that documents are related. I think you just mean other documents.

### PR DESCRIPTION
**Pull request by mattgarrish**
This pull request removes the recommendation about limiting access to persistent storage from "unrelated documents" since the unique origin requirement already prevents other epub publications from accessing this data. (Note that this deletes a test id. Not sure if the test would be useful to keep for the unique origin requirement?)

Although I think this is sufficient for reading systems, I've also added a new paragraph to the security section in the core specification to make epub creators aware that older reading systems are susceptible to exploits. It recommends not storing sensitive user data in persistent storage, but if there is no other option then the data should be encrypted to prevent trivial access to the information. Feedback welcome on whether this is sufficient, or if we should be saying something more or something else.

Fixes #2264

***

### :boom: Error: 500 Internal Server Error :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on May 22, 2022, 10:14 AM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [Spec Generator](https://www.w3.org/2015/labs/) - Spec Generator is the web service used to build specs that rely on ReSpec.

:link: [Related URL](https://labs.w3.org/spec-generator/?type=respec&url=https%3A%2F%2Fraw.githubusercontent.com%2Fw3c%2Fepub-specs%2F08e20987af5472c4adac0705e4cc866cbff52c35%2Fepub33%2Fcore%2Findex.html%3FisPreview%3Dtrue)

```

😭  Sorry, there was an error generating the HTML. Please report this issue!
Specification: http://labs.w3.org/spec-generator/uploads/fPU5fW?isPreview=true&publishDate=2022-05-22
ReSpec version: 32.1.6
File a bug: https://github.com/w3c/respec/
Error: Error: Evaluation failed: Timeout: document.respec.ready didn't resolve in 27624ms.
    at ExecutionContext._evaluateInternal (/u/spec-generator/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:221:19)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ExecutionContext.evaluate (/u/spec-generator/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:110:16)
    at async generateHTML (/u/spec-generator/node_modules/respec/tools/respecDocWriter.js:221:12)
    at async toHTML (/u/spec-generator/node_modules/respec/tools/respecDocWriter.js:92:18)
    at async Object.generate [as respec] (file:///u/spec-generator/generators/respec.js:15:44)
    at async file:///u/spec-generator/server.js:244:48

```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20w3c/epub-specs%232301.)._
</details>

### CODE REVIEW

**Comment by mattgarrish** [comment]
* EPUB 3.3:
    * [Preview](https://raw.githack.com/w3c/epub-specs/editorial/issue-2264/epub33/core/index.html)
    * [Diff](https://services.w3.org/htmldiff?doc1=https://labs.w3.org/spec-generator/%3Ftype=respec%26url=https://w3c.github.io/epub-specs/epub33/core/index.html&doc2=https://labs.w3.org/spec-generator/%3Ftype=respec%26url=https://raw.githack.com/w3c/epub-specs/editorial/issue-2264/epub33/core/index.html)
* EPUB Reading Systems 3.3:
    * [Preview](https://raw.githack.com/w3c/epub-specs/editorial/issue-2264/epub33/rs/index.html)
    * [Diff](https://services.w3.org/htmldiff?doc1=https://labs.w3.org/spec-generator/%3Ftype=respec%26url=https://w3c.github.io/epub-specs/epub33/rs/index.html&doc2=https://labs.w3.org/spec-generator/%3Ftype=respec%26url=https://raw.githack.com/w3c/epub-specs/editorial/issue-2264/epub33/rs/index.html)

File: epub33/core/index.html [hunk 1]
```diff
@@ -9423,10 +9423,20 @@ <h3>Recommendations</h3>
...
+				<p>EPUB creators also need to consider the inadvertent collection of information about users. Linking to
+					content on a publisher's web site, or remotely hosting resources on their servers, can lead to
 					profiling users, especially if unique tracking identifiers are added to the URLs.</p>
 
+				<p>When collecting and storing user information within an EPUB publication (e.g., through the use of <a
+						data-cite="html#dom-document-cookie">cookies</a> and <a data-cite="html#webstorage">web
+						storage</a> [[?html]]), EPUB creators need to consider to potential for data theft by other EPUB
+					publications on a reading system. Although [[epub-rs-33]] introduces a <a
+						data-cite="epub-rs-33#sec-container-iri">unique origin requirement</a> for EPUB publications,
+					which limits the potential for attacks, older reading systems may allow all EPUB publications access
```
Comment (dlazin) [inline epub33/core/index.html]: "older or non-conformant reading systems"?

File: epub33/core/index.html [hunk 2]
```diff
@@ -9423,10 +9423,20 @@ <h3>Recommendations</h3>
...
+					to the same persistent storage. Consequently, EPUB creators SHOULD NOT store sensitive user data in
+					persistent storage. If EPUB creators must store sensitive data, they SHOULD consider encrypting the
```
Comment (dlazin) [inline epub33/core/index.html]:
```suggestion
					persistent storage. If EPUB creators must store sensitive data, they SHOULD encrypt the
```

**Review by dlazin — APPROVED** [review]
I have removed the deleted MUST from tinyurl.com/epub-tests in anticipation of this being committed.

**Review by iherman — APPROVED** [review]

File: epub33/core/index.html (hunk 1 above)
Comment (mattgarrish) [reply epub33/core/index.html]: I've moved the examples to a parenthetical at the end to declutter a bit. It now says:

> Although [[epub-rs-33](#bib-epub-rs-33)] introduces a [unique origin requirement](https://www.w3.org/TR/epub-rs-33/#sec-container-iri) for EPUB publications, which limits the potential for attacks, there is still a risk that reading systems will allow EPUB publications access to shared persistent storage (e.g., older reading systems that have not been updated and non-conforming newer reading systems).

**Review by shiestyle — APPROVED** [review]

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2022-05-26-epub#section1-5) on 2022-05-26

List of resolutions:

- ***[Resolution No. 3](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2022-05-26-epub#resolution3):  Approve PR 2301, close issue 2264.***

<details><summary><i>View the transcript</i></summary>

#### 1.5. Persistent storage security.

_See github pull request [epub-specs#2301](https://github.com/w3c/epub-specs/pull/2301)._




_See github issue [epub-specs#2264](https://github.com/w3c/epub-specs/issues/2264)._




**Dave Cramer:** about unrelated documents.  

**Matt Garrish:** this is about two requirements that were still in the spec, but which are no longer applicable.  
… i.e., language around each document being treated as its own domain.  
… if you absolutely need to use persistent storage, then we recommend you encrypt instead of storing as plaintext.  

**Dave Cramer:** a lot of people have done proofs of concept of drafting epubs that can read data from local storage created by a different epub.  

**Matt Garrish:** not sure if javascript encrypting is trivial to break or not, but at least we are saying to pay attention to this.  
… we also recommend just not storing sensitive data in the first place, if you don't have to.  

> **Proposed resolution: Approve PR 2301, close issue 2264.** *(Wendy Reid)*

> *Dave Cramer:* +1.

> *Matt Garrish:* +1.

> *Brady Duga:* +1.

> *Shinya Takami (高見真也):* +1.

> *Wendy Reid:* +1.

> *Dan Lazin:* +1.

> *Toshiaki Koike:* +1.

> *Matthew Chan:* +1.

> *Masakazu Kitahara:* +1.

> ***Resolution #3: Approve PR 2301, close issue 2264.***

</details>

**Pull request merged**